### PR TITLE
Disallow single letter type variables for rule 2.6, change array flatten to joined

### PR DESCRIPTION
**Pull request by rkgibson2**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -211,10 +211,9 @@ class MyClassName {
 }
 ```
 
-* **2.6** For generics and associated types, use either a single capital letter or a `PascalCase` word that describes the generic. If this word clashes with a protocol that it conforms to or a superclass that it subclasses, you can append a `Type` suffix to the associated type or generic name.
+* **2.6** For generics and associated types, use either a `PascalCase` word that describes the generic. If this word clashes with a protocol that it conforms to or a superclass that it subclasses, you can append a `Type` suffix to the associated type or generic name.
```
Comment (nicksnyder) [inline README.md]: You need to delete the word "either"

**Comment by rkgibson2** [comment]
Oh yeah, thanks for catching that.

**Comment by cezarywojcik** [comment]
Please update the last updated date in the intro.

**Comment by rkgibson2** [comment]
I've updated the date, as well as addressing #13.

**Pull request merged**